sklearn/preprocessing/_encoders.py

```diff
@@ -586,77 +677,154 @@ class OrdinalEncoder(_BaseEncoder):
+        """Fit OneHotEncoder to X.
 
         Parameters
         ----------
-        X : array-like, shape [n_samples, n_features]
-            The data to determine the categories of each feature.
+        X : array-like of shape (n_samples, n_features)
+            The data to determine the categroies of each feature.
 
         Returns
         -------
         self
-
         """
         self._fit(X)
-
         return self
 
     def transform(self, X):
-        """Transform X to ordinal codes.
+        """Transform X to encoding
 
         Parameters
         ----------
-        X : array-like, shape [n_samples, n_features]
+        X : array-like of shape (n_samples, n_features)
             The data to encode.
 
         Returns
         -------
-        X_out : sparse matrix or a 2-d array
-            Transformed input.
+        X_out : ndarray of shape (n_samples, n_encoded_features)
+            Transformed array.
+        """
+        check_is_fitted(self)
+        X_list = self._check_X(X)
+        return self._transform_list(X_list)
+
+    def fit_transform(self, X, y=None):
+        """Fit encoder to X and transform X.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            The data to encode.
 
+        Returns
+        -------
+        X_out : ndarray of shape (n_samples, n_encoded_features)
+            Transformed array.
         """
-        X_int, _ = self._transform(X)
-        return X_int.astype(self.dtype, copy=False)
+        X_list = self._fit(X)
+        return self._transform_list(X_list)
 
-    def inverse_transform(self, X):
-        """Convert the data back to the original representation.
+
+class _SingleOrdinalEncoder(TransformerMixin, BaseEstimator):
+    """Ordinal Encoder for a single categorical feature."""
+    def __init__(self, categories='auto', dtype=np.float64, feature_idx=0):
+        self.categories = categories
+        self.dtype = dtype
+        self.feature_idx = feature_idx
+
+    def fit(self, X, y=None):
+        """Fit ordinal encoder on a single categorical feature.
 
         Parameters
         ----------
-        X : array-like or sparse matrix, shape [n_samples, n_encoded_features]
-            The transformed data.
+        X : ndarray of shape (n_samples,)
+            Categorical feature to encode.
 
         Returns
         -------
-        X_tr : array-like, shape [n_samples, n_features]
-            Inverse transformed array.
+        self
+        """
+        if isinstance(self.categories, str) and self.categories == 'auto':
+            cats = _encode(X)
+        else:  # categories were given
+            cats = np.array(self.categories, dtype=X.dtype)
+            if X.dtype != object and not np.all(np.sort(cats) == cats):
+                raise ValueError("Unsorted categories are not "
+                                 "supported for numerical categories")
+            diff = _encode_check_unknown(X, cats)
```

Comment:
Looking at the diff this is protected by `                if handle_unknown == 'error':` in master.

(Not sure if relevant)

Reply:
`_SingleOrdinalEncoder` will always error since it can not handle unknown (at the moment).